### DISCUSSION TITLE
HAL timebase in ST CUBE interfering with LPTIM-Tick?

### DISCUSSION THREAD

**Discussion by ufanders** [jefftenney/LPTIM-Tick#13] [Q&A]
Hi Jeff,

I've been able to integrate LPTIM-Tick into my project, but the system vectors off to Default_Handler, vector 0x0F (TIM3 ISR I think) right after _ISB() is called in vPortSuppressTicksAndSleep(). When configuring the SYS section in ST CUBE, it recommended to NOT use SysTick as the HAL timebase, so I chose TIM3 instead.

It seems like LPTIM1 runs the FreeRTOS timebase, while the ST HAL timebase is separate. I set configTICK_INTERRUPT_PRIORITY 4 per your instructions, because all application interrupt priorities in ST CUBE were 3 or below, but I still get hung up in Default_Handler.

What am I missing? My application would really benefit from lower-power sleep!

**Comment by jefftenney** [comment]
A few questions for you - 
Which STM32 are you using?  
Are you using STOP modes?
Can you put your project (or a minimal project that has the issue) on github?  I'd be happy to take a look.

**Reply by ufanders** [reply]
Hi Jeff, really cool of you to offer, thanks so much.

- I'm using an STM32L072CBT6.
- (Pre, Post)SleepProcessing() is undefined in my project.
- The repo and branch: 

[https://github.com/ufanders/retroblaze/tree/addLptimerTick/firmware/retroblazeFreeRtos](https://github.com/ufanders/retroblaze/tree/addLptimerTick/firmware/retroblazeFreeRtos)

Any tips are appreciated!

**Comment by jefftenney** [comment]
The vector at position 0xF is TIM2, not TIM3.  It looks like you are using TIM2 but have not registered an interrupt handler for it.  Seems like maybe this is a CubeMX code generation issue.
